source/beamline/AMBeamlineControlGroup.cpp

```diff
@@ -0,0 +1,62 @@
+#include "AMBeamlineControlGroup.h"
+#include "beamline/AMBeamlineControl.h"
+
+AMBeamlineControlGroup::AMBeamlineControlGroup(const QString &name, QObject *parent) :
+	AMConnectedControl(name, "", parent)
+{
+	status_ = new AMBeamlineControlGroupStatus(QString("%1%2").arg(name).arg("Status"), this);
+}
+
+AMBeamlineControlGroup::~AMBeamlineControlGroup()
+{
+
+}
+
+bool AMBeamlineControlGroup::isBad() const
+{
+	return status_->isBad();
+}
+
+bool AMBeamlineControlGroup::isGood() const
+{
+	return status_->isGood();
+}
+
+bool AMBeamlineControlGroup::hasControl(AMControl *control) const
+{
+	return status_->hasChildControl(control);
```

Comment:
It should be, at least how I've defined add/removeControl(). I'll change it to `children_->contains(control)`, as that's a little more clear.